ILTransform/ILRewriter.cs

```diff
@@ -417,8 +418,8 @@ private void RewriteFile(string source)
                 rewritten = true;
             }
 
-            Regex usingXunitRegex = new Regex(@"^\s*using\s+Xunit\s*(?:\\,*)?;");
-            bool usingXUnit = (_testProject.LastUsingLine >= 0 && lines.Take(_testProject.LastUsingLine).Any(l => usingXunitRegex.IsMatch(l)));
+            Regex usingXunitRegex = new Regex(@"^\s*using\s+Xunit\s*(?:\\.*)?;");
+            bool usingXUnit = (_testProject.LastUsingLine >= 0 && lines.Take(_testProject.LastUsingLine+1).Any(l => usingXunitRegex.IsMatch(l)));
```

Comment:
`LastUsingLine` is based on `line.StartsWith("using")` - not a very robust check and one of the things that this regex does better.  The potential problem being solved by the `Take` is that `using xx` could appear in the file (for example in a string literal).

I don't understand the `+1` here.  Do you have a case where there are "normal" usings but then the xunit one was missed when setting `LastUsingLine`?  Or is the line number off by one due to an insertion?  I'm asking here because `+1` seems like it might be covering a larger problem.

(Of course, the regex is more robust but still not an IL parser, which would be necessary to get this fully right, so we are approximating with whatever we do.)

Reply:
I understand most (or all) indexes are 1-indexed while the array of strings which contains the test file lines is 0-indexed. So the +1 is to contain the last using.

Reply:
It was adding "using Xunit;" when it was already in the file on jit/directed

Reply:
I think everything in ILTransform is 0-based.  It's all using that array (well, `List`) of strings.

So I think something unexplained was going on in your example.  AddILFactAttributes can modify the entrypoint before getting to this code, but I don't think that code is going to be before `using` where line number changes would matter.  Could `LastUsingLine` be incorrect?  It is simply using `line.StartsWith("using"), so whitespace or a comment at the beginning of the line would hide it.

Reply:
File lines are 0-indexed in the code. Let's say "using Xunit;" is on line 3, then Take(3) would inspect only lines 0, 1 and 2.